src/content/docs/en/reference/experimental-flags/sessions.mdx

```diff
@@ -229,12 +231,46 @@ export const onRequest = defineMiddleware(async (context, next) => {
 });
 ```
 
+### Session data types
+
+By default session data is untyped, and you can store arbitrary data in any key. Values are serialized and deserialized using [devalue](https://github.com/Rich-Harris/devalue), which is the same library used in content collections and actions. This means that supported types are the same, and include strings, numbers, `Date`, `Map`, `Set`, `URL`, arrays, and plain objects.
+
+You can optionally define TypeScript types for your session data by creating a `src/env.d.ts` file and adding a declaration for the `App.SessionData` type:
+
+```ts title="src/env.d.ts"
+declare namespace App {
+  interface SessionData {
+    user: {
+      id: string;
+      name: string;
+    };
+    cart: string[];
+  }
+}
+```
+
+This will allow you to access the session data with type-checking and auto-completion in your editor:
+
+```ts title="src/components/CartButton.astro"
+---
+const cart = await Astro.session.get('cart');
+// const cart: string[] | undefined
+
+const something = await Astro.session.get('something');
+// const something: any
+
+Astro.session.set('user', { id: 1, name: 'Houston' });
+// Error: Argument of type '{ id: number; name: string }' is not assignable to parameter of type '{ id: string; name: string; }'.
+---
+```
+:::caution
+
```

Comment:
```suggestion

:::caution
```
Just swapping the blank line here as it's not needed after the aside notation, and we usually *do* leave a line after a code sample!